### PR TITLE
feat(spanner): add support for float32

### PR DESCRIPTION
**Pull request by alkatrivedi**
This PR contains support for FLOAT32 type on Cloud Spanner.

### CODE REVIEW

File: src/codec.ts [hunk 1]
```diff
@@ -531,6 +550,7 @@ const TypeCode: {
   bool: 'BOOL',
   int64: 'INT64',
   pgOid: 'INT64',
+  float32: 'FLOAT32',
```
Comment (surbhigarg92) [inline src/codec.ts:553]: Should also add documentation in "paramtype", refer https://github.com/googleapis/nodejs-spanner/blob/main/src/codec.ts#L570

File: system-test/spanner.ts [hunk 2]
```diff
@@ -4863,6 +4863,235 @@ describe('Spanner', () => {
           });
         });
 
+        describe('float32', () => {
+          const float32Query = (done, database, query, value) => {
```
Comment (surbhigarg92) [inline system-test/spanner.ts:4867]: All the testcase are using float64 not float32. If param type is not passed then it take float64 as default. Add below for all tests
```
types: {
                v: 'float32',
              },
```

File: src/codec.ts (hunk 1 above)
Comment (alkatrivedi) [reply src/codec.ts:553]: Done

**Review by rahul2393 — APPROVED** [review]

File: system-test/spanner.ts [hunk 3]
```diff
@@ -4300,9 +4456,10 @@ describe('Spanner', () => {
       insertThenUpdateRow(done, postgreSqlTable);
     });
 
-    describe('insert & query', () => {
+    describe.skip('insert & query', () => {
```
Comment (surbhigarg92) [inline system-test/spanner.ts:4459]: Do not skip this, this is also used for other types. Instead comment the part where we are adding Float32 and add a comment for later

File: system-test/spanner.ts (hunk 2 above)
Comment (alkatrivedi) [reply system-test/spanner.ts:5042]: Done

File: system-test/spanner.ts (hunk 3 above)
Comment (alkatrivedi) [reply system-test/spanner.ts]: Done

**Review by surbhigarg92 — APPROVED** [review]

**Pull request merged**